### PR TITLE
[RDY] Avoid crash with `LC_ALL=C` and unicode filename

### LINKED ISSUE #1450: UnicodeEncodeError with os.path.isdir in NetworkManager.createRequest
Report with `LC_ALL=C`:

```
10:56:42 DEBUG    init       app:process_pos_args:275 Startup URL file:///tmp/people.inf.elte.hu/mersaai/helpvizsgahoz/cpp%20vizsg%ED%B3%83%ED%B2%A1ra/index.html
[...]
10:56:42 ERROR    misc       utils:wrapper:616 Error in qutebrowser.browser.network.networkmanager.NetworkManager.createRequest
Traceback (most recent call last):
  File "/usr/lib/python3.5/site-packages/qutebrowser/utils/utils.py", line 614, in wrapper
    return func(*args, **kwargs)
  File "/usr/lib/python3.5/site-packages/qutebrowser/browser/network/networkmanager.py", line 365, in createRequest
    op, req, outgoing_data)
  File "/usr/lib/python3.5/site-packages/qutebrowser/browser/network/filescheme.py", line 116, in createRequest
    if os.path.isdir(path):
  File "/usr/lib/python3.5/genericpath.py", line 42, in isdir
    st = os.stat(s)
UnicodeEncodeError: 'ascii' codec can't encode characters in position 55-60: ordinal not in range(128)
```

### PR DESCRIPTION
**Pull request by 7lb**
Fixes #1450 

This fixes the crash, instead displaying an error.

It's not the best option, but it's better than a crash

---
This change is [Reviewable](https://reviewable.io/reviews/qutebrowser/qutebrowser/3175)

### CODE REVIEW

File: qutebrowser/browser/webkit/network/filescheme.py [hunk 1]
```diff
@@ -127,7 +127,10 @@ def createRequest(self, _op, request, _outgoing_data):
...
-            data = dirbrowser_html(path)
-            return networkreply.FixedDataNetworkReply(
-                request, data, 'text/html', self.parent())
+        try:
+            if os.path.isdir(path):
+                data = dirbrowser_html(path)
+                return networkreply.FixedDataNetworkReply(
+                    request, data, 'text/html', self.parent())
+        except UnicodeEncodeError:
+            return None
```
Comment (jgkamat) [inline qutebrowser/browser/webkit/network/filescheme.py]: I think that one of the tests is failing because there's no test for the case you added. I'm not sure if there's a way to trigger this from the test suite, but if you think of one, could you add a unit test for this?

Comment (7lb) [reply qutebrowser/browser/webkit/network/filescheme.py]: Mmm, I did not notice any failing errors (not more than those that normally fail at least).

I ran `tox -e py36 tests/unit/browser/webkit/network/test_filescheme.py` and had 25 passing tests.

I modified `test_file` in `test_filescheme.py` to use `φτωχικὸ` as a filename instead of `foo`, but that will not raise an exception in tox even with `LC_ALL=C` so I'm not sure what option there are to test this properly, any ideas?

Comment (jgkamat) [reply qutebrowser/browser/webkit/network/filescheme.py]: Sorry, I was a bit too vauge there, the coverage tests are failing (meaning that coverage dropped on a perfect file) :P. 

I actually can't test/verify anything about the webkit crash, I don't have webkit installed. Could you please provide a webkit crash stacktrace for me to look at?

The reason I'm asking is because (like you said) the test:

```
    def test_file(self, tmpdir):
        filename = tmpdir / 'φτωχικὸ'
        os.environ['LC_ALL'] = "C"
        filename.ensure()
        url = QUrl.fromLocalFile(str(filename))
        req = QNetworkRequest(url)
        handler = filescheme.FileSchemeHandler(win_id=0)
        reply = handler.createRequest(None, req, None)
        assert reply is None
```

runs the changed code for me fine, without a patch. I'm not sure if setting the env var in python is sufficient or not (or if I need to be running all of qb, not just the test environment). A stacktrace would help with that :)

Comment (7lb) [reply qutebrowser/browser/webkit/network/filescheme.py]: Here's a stacktrace, it's actually the same as that in #1450 :)
```
$ LC_ALL=C python -m qutebrowser --temp-basedir --backend webkit xxφτωχικὸ 
21:31:43 ERROR: Error in qutebrowser.browser.webkit.network.networkmanager.NetworkManager.createRequest
Traceback (most recent call last):
  File "/home/l/.proj/qutebrowser/qutebrowser/utils/utils.py", line 635, in wrapper
    return func(*args, **kwargs)
  File "/home/l/.proj/qutebrowser/qutebrowser/browser/webkit/network/networkmanager.py", line 390, in createRequest
    op, req, outgoing_data)
  File "/home/l/.proj/qutebrowser/qutebrowser/browser/webkit/network/filescheme.py", line 130, in createRequest
    if os.path.isdir(path):
  File "/usr/lib/python3.6/genericpath.py", line 42, in isdir
    st = os.stat(s)
UnicodeEncodeError: 'ascii' codec can't encode characters in position 28-72: ordinal not in range(128)
```

In any case yeah, I hadn't thought about coverage, in that case adding a tests for an unicode filename is useful even if, de facto, it doesn't work

Comment (7lb) [reply qutebrowser/browser/webkit/network/filescheme.py]: Actually, I take that back. Since the exception is never raised during tests, those two lines are never tested against.

Comment (The-Compiler) [reply qutebrowser/browser/webkit/network/filescheme.py]: Looks like @lejar volunteered to write a test in #1450, which would probably catch this too :wink:

Comment (7lb) [reply qutebrowser/browser/webkit/network/filescheme.py]: Great! Feel free to ping me should I need to no more for this PR
